internal/backends/winit/event_loop.rs

```diff
@@ -25,94 +25,8 @@ use winit::event::WindowEvent;
-        {
-            #[cfg(feature = "wayland")]
-            {
-                use winit::platform::wayland::EventLoopBuilderExtWayland;
-                builder.with_any_thread(true);
-            }
-            #[cfg(feature = "x11")]
-            {
-                use winit::platform::x11::EventLoopBuilderExtX11;
-                builder.with_any_thread(true);
-
-                // Under WSL, the compositor sometimes crashes. Since we cannot reconnect after the compositor
-                // was restarted, the application panics. This does not happen when using XWayland. Therefore,
-                // when running under WSL, try to connect to X11 instead.
-                #[cfg(feature = "wayland")]
-                if std::fs::metadata("/proc/sys/fs/binfmt_misc/WSLInterop").is_ok()
-                    || std::fs::metadata("/run/WSL").is_ok()
-                {
-                    builder.with_x11();
-                }
-            }
-        }
-        #[cfg(target_family = "windows")]
-        {
-            use winit::platform::windows::EventLoopBuilderExtWindows;
-            builder.with_any_thread(true);
-        }
-
-        let instance =
-            builder.build().map_err(|e| format!("Error initializing winit event loop: {e}"))?;
 
-        Ok(Self { instance })
-    }
-}
-
-pub(crate) struct RunningEventLoop<'a> {
-    active_event_loop: &'a ActiveEventLoop,
-}
-
-pub(crate) enum ActiveOrInactiveEventLoop<'a> {
-    #[allow(unused)]
-    Active(&'a ActiveEventLoop),
-    #[allow(unused)]
-    Inactive(&'a winit::event_loop::EventLoop<SlintEvent>),
-}
-
-pub(crate) trait EventLoopInterface {
-    fn create_window(
-        &self,
-        window_attributes: winit::window::WindowAttributes,
-    ) -> Result<winit::window::Window, winit::error::OsError>;
-    #[allow(unused)]
-    fn event_loop(&self) -> ActiveOrInactiveEventLoop<'_>;
-}
-
-impl EventLoopInterface for NotRunningEventLoop {
-    fn create_window(
-        &self,
-        window_attributes: winit::window::WindowAttributes,
-    ) -> Result<winit::window::Window, winit::error::OsError> {
-        #[allow(deprecated)]
-        self.instance.create_window(window_attributes)
-    }
-    fn event_loop(&self) -> ActiveOrInactiveEventLoop<'_> {
-        ActiveOrInactiveEventLoop::Inactive(&self.instance)
-    }
-}
-
-impl EventLoopInterface for RunningEventLoop<'_> {
-    fn create_window(
-        &self,
-        window_attributes: winit::window::WindowAttributes,
-    ) -> Result<winit::window::Window, winit::error::OsError> {
-        self.active_event_loop.create_window(window_attributes)
-    }
-    fn event_loop(&self) -> ActiveOrInactiveEventLoop<'_> {
-        ActiveOrInactiveEventLoop::Active(self.active_event_loop)
-    }
-}
-
-scoped_tls_hkt::scoped_thread_local!(pub(crate) static CURRENT_WINDOW_TARGET : for<'a> &'a RunningEventLoop<'a>);
+scoped_tls_hkt::scoped_thread_local!(pub(crate) static ACTIVE_EVENT_LOOP : for<'a> &'a ActiveEventLoop);
```

Comment:
I might be wrong, but i'm thinking we don't need this global and the proxy. We could just always add the window we want to show the the inactive_windows list so we can show them on the next iteration, right?

Thinking about it, the problem would be about how to best report error, but the error would be thrown from run_event_loop() instead of show(), which is not the best, but probably would still works.

Reply:
That's an interesting idea. Yes, the main issue is that functions on the window adapter need to access the active event loop to create windows, but delaying that to the next event loop cycle sounds feasible. I'll give it a try.

Reply:
(I'll take care of this in a follow-up)